api/v1beta1/azuremanagedcontrolplane_default.go

```diff
@@ -101,8 +101,13 @@ func (m *AzureManagedControlPlane) setDefaultSubnet() {
 // setDefaultFleetsMember sets the default FleetsMember for an AzureManagedControlPlane.
 func setDefaultFleetsMember(fleetsMember *FleetsMember, labels map[string]string) *FleetsMember {
 	result := fleetsMember.DeepCopy()
-	if clusterName, ok := labels[clusterv1.ClusterNameLabel]; ok && fleetsMember != nil && fleetsMember.Name == "" {
-		result.Name = clusterName
+	if fleetsMember != nil {
+		if clusterName, ok := labels[clusterv1.ClusterNameLabel]; ok && fleetsMember.Name == "" {
```

Comment:
Do we wanna also make sure that clusterName is not empty string before we make this assignment?

For example:

https://play.golang.com

Reply:
It seems like other uses of getting the cluster name via `clusterv1.ClusterNameLabel` assume the clusterName is not empty. Since we're checking the label exists, I think the clusterName can't be empty. 

https://github.com/willie-yao/cluster-api-provider-azure/blob/451b397c4a3b8fc23e8c238a737703ca5a587c08/api/v1beta1/azuremachine_default.go#L232